test/jdk/java/net/httpclient/HttpResponseLimitingTest.java

```diff
@@ -0,0 +1,396 @@
+            } catch (IOException exception) {
+                throw new UncheckedIOException(exception);
+            }
+        }
+
+        private ServerClientPair {
+            server.start();
+        }
+
+        private static ServerClientPair of(HttpClient.Version version, boolean secure) {
+
+            // Create the server and the request URI
+            SSLContext sslContext = secure ? SSL_CONTEXT : null;
+            HttpTestServer server = createServer(version, sslContext);
+            String handlerPath = "/";
+            String requestUriScheme = secure ? "https" : "http";
+            URI requestUri = URI.create(requestUriScheme + "://" + server.serverAuthority() + handlerPath);
+
+            // Register the request handler
+            server.addHandler(
+                    (exchange) -> {
+                        exchange.getResponseHeaders().addHeader(RESPONSE_HEADER_NAME, RESPONSE_HEADER_VALUE);
+                        exchange.sendResponseHeaders(200, RESPONSE_BODY.length);
+                        try (var outputStream = exchange.getResponseBody()) {
+                            outputStream.write(RESPONSE_BODY);
+                        }
+                        exchange.close();
+                    },
+                    handlerPath);
+
+            // Create the client and the request
+            HttpClient client = createClient(version, sslContext);
+            HttpRequest request = HttpRequest.newBuilder(requestUri).version(version).build();
+
+            // Create the pair
+            return new ServerClientPair(server, client, request);
+
+        }
+
+        private static HttpTestServer createServer(HttpClient.Version version, SSLContext sslContext) {
+            try {
+                return HttpTestServer.create(version, sslContext);
+            } catch (IOException exception) {
+                throw new UncheckedIOException(exception);
+            }
+        }
+
+        private static HttpClient createClient(HttpClient.Version version, SSLContext sslContext) {
+            HttpClient.Builder builder = HttpClient.newBuilder().version(version).proxy(NO_PROXY);
+            if (sslContext != null) {
+                builder.sslContext(sslContext);
+            }
+            return builder.build();
+        }
+
+        private <T> HttpResponse<T> request(BodyHandler<T> downstreamHandler, long capacity) throws Exception {
+            var handler = BodyHandlers.limiting(downstreamHandler, capacity);
+            return client.send(request, handler);
+        }
+
+        @Override
+        public String toString() {
+            String version = client.version().toString();
+            return client.sslContext() != null ? version.replaceFirst("_", "S_") : version;
+        }
+
+    }
+
+    @AfterAll
+    static void closeServerClientPairs() {
+        for (var pair : new ServerClientPair[]{HTTP1, HTTPS1, HTTP2, HTTPS2}) {
+            pair.client.close();
+            pair.server.stop();
```

Comment:
I think we should consider catching any exception that might happen here when closing each client/server and log a message and continue closing the next pair, especially since this is not a `othervm` test - we wouldn't want the sockets to continue to be open after this test completes.

Reply:
Implemented in 085e48e8b54475e630674686a2c884b241565b09.